packages/block-library/src/navigation-link/edit.js

```diff
@@ -209,6 +209,34 @@ export const updateNavigationLinkBlockAttributes = (
 	} );
 };
 
+const useIsInvalidLink = ( kind, type, id ) => {
+	const isPostType = kind === 'post-type' || type === 'post';
```

Comment:
I think we'll also want to check for type of 'page'. This issue would only show up for folks with old data since kind is not set.

```jsx
const isPostType = kind === 'post-type' || type === 'post' || type = 'page';
```

It looks like we should be mirroring checks in https://github.com/WordPress/gutenberg/blob/efc28f91fb2c4924b173bc7c8f60dbffac62a7d2/packages/block-library/src/navigation-link/index.php#L107-L112

Reply:
Those are exactly the checks I meant to mirror, must have missed that one. Thanks!